test/tla/Bug931.tla

```diff
@@ -3,6 +3,6 @@ EXTENDS  FiniteSets
 
 Init == TRUE
 Next == TRUE
-Inv == Cardinality({}) = 0
+Inv == Cardinality({}) /= 1
```

Comment:
The expression optimizer transforms `Cardinality(S) = 0` into `S = {}` so, in this case, `Cardinality({}) = 0` into `{} = {}`. The new more generic simplification rule for equality transforms `{} = {}` into `TRUE`, and therefore suppresses the bug this test is trying to reproduce. I understand that suppressing the polymorphic type error here is a good thing, therefore I changed the test instead of changing the simplification behavior. Let me know if you guys disagree, please.